datafusion/physical-expr/src/hash_utils.rs

```diff
@@ -68,15 +68,15 @@ hash_value!(i8, i16, i32, i64, i128, i256, u8, u16, u32, u64);
 hash_value!(bool, str, [u8]);
 
 macro_rules! hash_float_value {
-    ($($t:ty),+) => {
+    ($(($t:ty, $i:ty)),+) => {
         $(impl HashValue for $t {
             fn hash_one(&self, state: &RandomState) -> u64 {
-                state.hash_one(self.to_le_bytes())
+                state.hash_one(<$i>::from_ne_bytes(self.to_ne_bytes()))
```

Comment:
This is a drive by fix that improves hashing performance of floating points. This is because there are specialized methods for hashing `i32`, `i64` as opposed to `[u8;N]`

Reply:
Looking at this now, are we correctly handling NaNs here?
Looking at the ordered-float implementation for `Hash`, it first normalizes nans https://docs.rs/ordered-float/3.3.0/src/ordered_float/lib.rs.html#160

Reply:
See https://github.com/apache/arrow-datafusion/issues/4051

TLDR OrderedFloat is a more strict ordering than we need

Reply:
👍

Reply:
I know it is a pain, but i would recommend pulling this into its own PR

Reply:
This doesn't materially change the hashing? It was a bitwise hash before and after, all that changes is it hashes an integer comprised of the bytes instead of an array of bytew